test/apps/mmap/mmap_vmrss.c

```diff
@@ -0,0 +1,154 @@
+// SPDX-License-Identifier: MPL-2.0
+
+#define _GNU_SOURCE
+
+#include "../network/test.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <sys/mman.h>
+#include <fcntl.h>
+
+#define PAGE_SIZE 4096
+#define NUM_PAGES 1024
+#define TOTAL_SIZE (PAGE_SIZE * NUM_PAGES)
+
+typedef enum rss_type {
+	anon,
+	file,
+	total,
+} rss_type;
+
+long get_vm_rss_kb(rss_type type)
+{
+	pid_t pid = getpid();
+	char path[64];
+	snprintf(path, sizeof(path), "/proc/%d/status", pid);
+
+	FILE *f = fopen(path, "r");
+	if (!f) {
+		perror("fopen /proc/[pid]/status");
+		exit(1);
+	}
+
+	char line[256];
+	long rss_kb = -1;
+	const char *target_field = NULL;
+	switch (type) {
+	case anon:
+		target_field = "RssAnon:";
+		break;
+	case file:
+		target_field = "RssFile:";
+		break;
+	case total:
+		target_field = "VmRSS:";
+		break;
+	default:
+		perror("Unknown rss_type\n");
+		exit(1);
+	}
+
+	while (fgets(line, sizeof(line), f)) {
+		if (strncmp(line, target_field, strlen(target_field)) == 0) {
+			sscanf(line + strlen(target_field), "%ld", &rss_kb);
+			break;
+		}
+	}
+
+	fclose(f);
+
+	if (rss_kb < 0) {
+		fprintf(stderr, "Failed to parse VmRSS\n");
+		exit(1);
+	}
+
+	return rss_kb;
+}
+
+FN_TEST(rss_anon)
+{
+	void *mem = mmap(NULL, TOTAL_SIZE, PROT_READ | PROT_WRITE,
+			 MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
+	if (mem == MAP_FAILED) {
+		perror("mmap");
+		exit(1);
+	}
+
+	long rss_anon_before, rss_file_before, rss_before;
+
+	// The first call to `TEST_SUCC` and `get_vm_rss_kb()` may trigger
+	// lazy mapping of additional pages, such as shared libraries or files.
+	// These pages are not counted in RSS until they are actually accessed.
+	TEST_SUCC(get_vm_rss_kb(anon));
+
+	rss_anon_before = get_vm_rss_kb(anon);
+	rss_file_before = get_vm_rss_kb(file);
+	rss_before = get_vm_rss_kb(total);
```

Comment:
Be consistent:
 - If you decide to use `TEST_SUCC` on `get_vm_rss_kb`, use it for all `get_vm_rss_kb`s instead of just for the first one;
 - The ways that you declare variables in `FN_TEST(rss_anon)` and `FN_TEST(rss_file)` are different, but it's better to unify it.
```c
FN_TEST(rss_anon)
{
	TEST_SUCC(get_vm_rss_kb(anon));

	long rss_anon_before = TEST_SUCC(get_vm_rss_kb(anon));
	long rss_file_before = TEST_SUCC(get_vm_rss_kb(file));
	long rss_before = TEST_SUCC(get_vm_rss_kb(total));
}

FN_TEST(rss_file)
{
	TEST_SUCC(get_vm_rss_kb(anon));

	long rss_anon_before = TEST_SUCC(get_vm_rss_kb(anon));
	long rss_file_before = TEST_SUCC(get_vm_rss_kb(file));
	long rss_before = TEST_SUCC(get_vm_rss_kb(total));
}
```